adal/src/main/java/com/microsoft/aad/adal/ADALError.java

```diff
@@ -120,6 +120,11 @@
     SERVER_ERROR("Server returned an error"),
 
     /**
+     * Server returned the 500/503/504 error.
+     */
+    SERVER_ERROR_FOR_RETRY("Server returned the 500/503/504 error"),
```

Comment:
hm, retry is our internal interpretation for these three error code, i don't think we should name it this way. @addev-ashish What do you think?